packages/@n8n/task-runner/src/js-task-runner/js-task-runner.ts

```diff
@@ -105,6 +105,43 @@ export class JsTaskRunner extends TaskRunner {
 		});
 	}
 
+	private preventPrototypePollution() {
+		if (process.env.NODE_ENV === 'test') return; // needed for Jest
+
+		[
+			Object,
+			Array,
+			Buffer,
+			Function,
+			RegExp,
+			Error,
+			Number,
+			String,
+			Promise,
+			Date,
+			Map,
+			Set,
+			BigInt,
+			Symbol,
+			WeakMap,
+			WeakSet,
+			Int8Array,
+			Uint8Array,
+			Int16Array,
+			Uint16Array,
+			Int32Array,
+			Uint32Array,
+			Float32Array,
+			Float64Array,
+		]
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+			.map((constructor) => constructor.prototype)
+			.forEach(Object.freeze);
+
+		Object.setPrototypeOf = () => false;
+		Reflect.setPrototypeOf = () => false;
```

Comment:
We should also do this within the task context. We don't need to provide these to the execution context. And also block `Object.getPrototypeOf`

Reply:
Yes, like that 👌